### PR TITLE
Fixed NPE in Precondition

### PR DESCRIPTION
**Pull request by ben-Draeger**
The DescriptionModificationUptPrecondition (amongst others) threw a NullPointerException when encountering a DescriptionModificationReport containing a ReportPart without a ModificationType attribute.

This should fix the problem.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Adherence to coding conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Adherence to javadoc conventions
  * [x] Pull Request Assignee
  * [x] Reviewer
* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* config update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* SDCcc executable ran against a test device (if necessary)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

**Review by maximilianpilz — CHANGES_REQUESTED** [review]
please fix the build, it failed, and change the copyright notice in the license headers of the files which you changed

**Comment by ben-Draeger** [comment]
The licence header check fails because I updated the license header in the 2 files I touched.
This can be rectified by merging in the change from PR #134, which introduced a more powerful github action for licence header checking that can deal with this type of headers.

**Review by maximilianpilz — CHANGES_REQUESTED** [review]
please make use of the new license check from the main branch

**Comment by ben-Draeger** [comment]
The Licence check was merged in from the main branch and it seems to work fine.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - autocommit of the database being on, it is now always off
 - local address resolver sometimes giving a wrong address
 - local address resolver doing a probe which may not be tolerated by some peers
+- potential NullPointerException in DescriptionModificationUptPrecondition.
```
Comment (maximilianpilz) [inline CHANGELOG.md:20]:
```suggestion
- potential NullPointerException in DescriptionModificationUptPrecondition
```

**Review by maximilianpilz — CHANGES_REQUESTED** [review]

**Pull request merged**